.devcontainer/configure.py

```diff
@@ -0,0 +1,153 @@
+  else:
+    output = subprocess.check_output(cmd, stderr=stderr)
+  return output.decode('UTF-8').strip()
+
+
+def is_nvidia_gpu_available():
+  """This function verifies NVIDIA Docker runtime is installed and
+  available. It also checks NVIDIA GPU are available and the driver
+  is installed."""
+
+  docker_executable = which('docker')
+  if docker_executable is None:
+    raise RuntimeError("Docker is not installed or in the user path.")
+
+  command_str = f"'{docker_executable}' info --format '{{{{json .}}}}'"
+  data = json.loads(run_shell(command_str))
+
+  if "nvidia" not in data["Runtimes"].keys():
+    return (
+        False, "NVIDIA Docker Runtime is not available. Please see: "
+        "https://developer.nvidia.com/nvidia-container-runtime for additional "
+        "information.")
+
+  nvidia_smi_executable = which('nvidia-smi')
+  if nvidia_smi_executable is None:
+    return False, "NVIDIA Driver is not installed or not in the user path."
+
+  command_str = f"{nvidia_smi_executable} --query-gpu=gpu_name --format=csv"
+
+  if run_shell(command_str).splitlines()[1:]:  # Skip header
+    return True, None
+  else:
+    return False, "No NVIDIA GPU was found on this system."
+
+
+def get_input(question, default_answer='', accepted_answers=None):
+  accepted_answers = [x.strip().lower() for x in accepted_answers]
+
+  while True:
+
+    try:
+      answer = input(f"{question} ")  # pylint: disable=bad-builtin
+    except EOFError:
+      answer = default_answer
+    answer = (answer or default_answer).strip().lower()
+
+    if answer not in accepted_answers:
+      print(
+          f"Unsupported answer received: {answer}. Expected: {accepted_answers}"
+      )
+      continue
+
+    break
+
+  return answer.strip().lower()
+
+
+if __name__ == "__main__":
+
+  if os.path.isfile(os.path.join(CURRENT_DIR, "docker-compose.yml")):
+    if get_input(
+        "A `docker-compose.yml` already exists. Are you certain you want to overwrite it [y/N]?",
+        default_answer="n",
+        accepted_answers=["y", "n"]) == "n":
+      sys.exit(1)
+
+  with open(os.path.join(CURRENT_DIR, "docker-compose.base.yml")) as f:
+    docker_config = json.load(f)
+
+  use_official_img = get_input(
+      "Do you wish to use the official prebuild development containers [Y/n]?",
+      default_answer="y",
+      accepted_answers=["y", "n"]) == "y"
+
+  use_nvidia_gpu = False
+  nvidia_gpu_available, err_msg = is_nvidia_gpu_available()
+  if nvidia_gpu_available:
+    use_nvidia_gpu = get_input("Do you wish to use NVIDIA GPU [y/N]?",
+                               default_answer="n",
+                               accepted_answers=["y", "n"]) == "y"
+  else:
+    print(f"NVIDIA GPUs are not available for use: {err_msg}")
+
+  docker_img_shortname = DOCKER_IMAGE_SHORTNAME_DICT[
+      "nvidia" if use_nvidia_gpu else "base"]
+
+  if use_official_img:
+    del docker_config["services"]["iree-dev"]["build"]
+
+    docker_iree_registry = SourceFileLoader(
```

Comment:
*sigh* oh Python

Reply:
yeah unfortunately ... Not like you can do out of tree imports in any other way.
You can also add the directory to PYTHONPATH but it's not any nicer